pkg/networkutils/ipgenerator.go

```diff
@@ -35,10 +39,9 @@ func (ipgen IPGenerator) GenerateUniqueIP(cidrBlock string) (string, error) {
 
 // generates a random ip within the specified cidr block
 func (ipgen IPGenerator) randIp(cidr *net.IPNet) (net.IP, error) {
-	rand.Seed(time.Now().UnixNano())
 	newIp := *new(net.IP)
 	for i := 0; i < 4; i++ {
-		newIp = append(newIp, byte(rand.Intn(256))&^cidr.Mask[i]|cidr.IP[i])
+		newIp = append(newIp, uint8(ipgen.rand.Intn(256))&^cidr.Mask[i]|cidr.IP[i])
```

Comment:
What's the reason for this? byte should be type-defined to uint8

Reply:
Left over from a small experiment, sorry. Thanks!